### PR TITLE
[AArch64] Add the Ampere1B core

### PR DESCRIPTION
**Pull request by ptomsich**
The Ampere1B is Ampere's third-generation core implementing a superscalar, out-of-order microarchitecture with nested virtualization, speculative side-channel mitigation and architectural support for defense against ROP/JOP style software attacks.

Ampere1B is an ARMv8.7+ implementation, adding support for the FEAT WFxT, FEAT CSSC, FEAT PAN3 and FEAT AFP extensions.  It also includes all features of the second-generation Ampere1A, such as the Memory Tagging Extension and SM3/SM4 cryptography instructions.

This provides basic enablement (reusing the scheduling model from Ampere1).
An updated scheduling model will be added as a separate commit.

### CODE REVIEW

**Comment by ptomsich** [comment]
The clang-format failures are caused by preexisting/unchanged code.
All newly added code passes clang-format.

File: llvm/include/llvm/TargetParser/AArch64TargetParser.h [hunk 1]
```diff
@@ -784,27 +784,32 @@ inline constexpr CpuInfo CpuInfos[] = {
      (AArch64::ExtensionBitset(
          {AArch64::AEK_AES, AArch64::AEK_SHA2, AArch64::AEK_CRC}))},
     {"tsv110", ARMV8_2A,
-     (AArch64::ExtensionBitset(
-         {AArch64::AEK_AES, AArch64::AEK_SHA2, AArch64::AEK_DOTPROD,
-          AArch64::AEK_FP16, AArch64::AEK_FP16FML, AArch64::AEK_PROFILE,
-          AArch64::AEK_JSCVT, AArch64::AEK_FCMA}))},
+     (AArch64::ExtensionBitset({AArch64::AEK_AES, AArch64::AEK_SHA2,
```
Comment (jthackray) [inline llvm/include/llvm/TargetParser/AArch64TargetParser.h:787]: Nit: this appears to be a NFC whitespace cleanup for an unrelated core. Would prefer it if your change was just for ampere1b.

(Can we clang-format in a later change, as there are a lot of other places indentation needs fixing?)

**Review by jthackray — CHANGES_REQUESTED** [review]

**Comment by jthackray** [comment]
> The clang-format failures are caused by preexisting/unchanged code. All newly added code passes clang-format.

Yes, I had this issue when I landed new cores previously. This whole file needs an NFC clang-format cleanup, IMHO.

File: llvm/include/llvm/TargetParser/AArch64TargetParser.h (hunk 1 above)
Comment (ptomsich) [reply llvm/include/llvm/TargetParser/AArch64TargetParser.h:787]: Looks like that one slipped it when adding the clang-format changes for the new core.
Thanks for the heads-up, I'll update the PR.

Comment (ptomsich) [reply llvm/include/llvm/TargetParser/AArch64TargetParser.h]: This indeed was a missed 's'plit while performing an interactive add.
PR updated.

**Review by jthackray — APPROVED** [review]
LGTM

**Pull request merged**